crates/bevy_render/src/texture/compressed_image_saver.rs

```diff
@@ -19,46 +19,48 @@ impl AssetSaver for CompressedImageSaver {
     type OutputLoader = ImageLoader;
     type Error = CompressedImageSaverError;
 
-    fn save<'a>(
+    async fn save<'a>(
         &'a self,
         writer: &'a mut bevy_asset::io::Writer,
         image: SavedAsset<'a, Self::Asset>,
         _settings: &'a Self::Settings,
-    ) -> bevy_utils::BoxedFuture<'a, Result<ImageLoaderSettings, Self::Error>> {
-        // PERF: this should live inside the future, but CompressorParams and Compressor are not Send / can't be owned by the BoxedFuture (which _is_ Send)
-        let mut compressor_params = basis_universal::CompressorParams::new();
-        compressor_params.set_basis_format(basis_universal::BasisTextureFormat::UASTC4x4);
-        compressor_params.set_generate_mipmaps(true);
+    ) -> Result<ImageLoaderSettings, Self::Error> {
         let is_srgb = image.texture_descriptor.format.is_srgb();
-        let color_space = if is_srgb {
-            basis_universal::ColorSpace::Srgb
-        } else {
-            basis_universal::ColorSpace::Linear
+
+        let compressed_basis_data = {
+            // PERF: this should live inside the future, but CompressorParams and Compressor are not Send / can't be owned by the BoxedFuture (which _is_ Send)
```

Comment:
We can remove this as this does live "inside" the future now. Intersting that this pattern works!

Reply:
Opps nice catch! And yeah annoying that the extra scope is needed [(known problem](https://rust-lang.github.io/async-book/07_workarounds/03_send_approximation.html) appearantly) but is cool that it works at all! Maybe future borrow checker one day :)